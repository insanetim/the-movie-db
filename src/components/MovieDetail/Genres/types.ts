import { IMovieDetail } from 'src/interfaces/movie.interface'

export interface GenresProps {
  genres: IMovieDetail['genres']
}
