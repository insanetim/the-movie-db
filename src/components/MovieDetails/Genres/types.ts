import { IMovieDetails } from 'src/interfaces/movie.interface'

export type GenresProps = {
  genres: IMovieDetails['genres']
}
