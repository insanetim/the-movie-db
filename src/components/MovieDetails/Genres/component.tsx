import { Col, Flex, Tag, Typography } from 'antd'

import { GenresProps } from './types'

const Genres: React.FC<GenresProps> = ({ genres }) => (
  <Col span={24}>
    <Typography.Paragraph>
      <b>Genres: </b>
      <Flex
        gap={4}
        style={{ display: 'inline-flex' }}
      >
        {genres.map(({ id, name }) => (
          <Tag key={id}>{name.toUpperCase()}</Tag>
        ))}
      </Flex>
    </Typography.Paragraph>
  </Col>
)

export default Genres
