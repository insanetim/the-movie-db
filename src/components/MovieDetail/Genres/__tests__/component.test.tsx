import { render } from '@testing-library/react'

import Genres from '../component'

describe('Genres component', () => {
  it('should match snapshot', () => {
    const { asFragment } = render(
      <Genres genres={[{ id: 1, name: 'test/genre' }]} />
    )

    expect(asFragment()).toMatchSnapshot()
  })
})
