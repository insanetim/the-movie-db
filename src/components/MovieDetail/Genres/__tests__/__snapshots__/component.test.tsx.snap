// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Genres component should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="ant-col ant-col-24 css-dev-only-do-not-override-1rqnfsa"
  >
    <div
      class="ant-typography css-dev-only-do-not-override-1rqnfsa"
    >
      <b>
        Genres: 
      </b>
      <span
        class="ant-tag css-dev-only-do-not-override-1rqnfsa"
      >
        TEST/GENRE
      </span>
    </div>
  </div>
</DocumentFragment>
`;
